integration/querydsl/blaze-persistence-querydsl-testsuite/src/test/java/com/blazebit/persistence/querydsl/BasicQueryTest.java

```diff
@@ -0,0 +1,744 @@
+                    .select(idHolderCTE.id).from(idHolderCTE)
+                    .fetch();
+
+            assertNotNull(fetch);
+        });
+    }
+
+    @Test
+    @Category({ NoDatanucleus.class, NoEclipselink.class, NoOpenJPA.class, NoMySQLOld.class })
+    public void testCTEWithBinds2() {
+        doInJPA(entityManager -> {
+            List<Long> fetch = new BlazeJPAQuery<Document>(entityManager, cbf)
+                .with(idHolderCTE, select(
+                        bind(idHolderCTE.id, document.id)).from(document))
+                .select(idHolderCTE.id).from(idHolderCTE)
+                .fetch();
+
+                assertNotNull(fetch);
+        });
+    }
+
+    @Test
+    @Category({ NoDatanucleus.class, NoEclipselink.class, NoOpenJPA.class, NoMySQL.class })
+    public void testCTEUnion() {
+        doInJPA(entityManager -> {
+            List<Long> fetch = new BlazeJPAQuery<Document>(entityManager, cbf)
+                    .with(idHolderCTE, idHolderCTE.id).as(union(select(document.id).from(document), intersect(select(document.id).from(document), select(document.id).from(document))))
+                    .select(idHolderCTE.id).from(idHolderCTE)
+                    .fetch();
+
+            assertNotNull(fetch);
+        });
+    }
+
+
+
+    @Test
+    @Category({ NoDatanucleus.class, NoEclipselink.class, NoOpenJPA.class, NoMySQLOld.class })
+    public void testCTEFromValues() {
+        doInJPA(entityManager -> {
+            Document theBook = new Document();
+            theBook.setId(1337L);
+            theBook.setName("test");
+
+            List<Long> fetch = new BlazeJPAQuery<Document>(entityManager, cbf)
+                    .with(idHolderCTE, idHolderCTE.id).as(new BlazeJPAQuery<Document>(entityManager, cbf)
+                            .fromValues(document, Collections.singleton(theBook))
+                            .select(document.id))
+                    .select(idHolderCTE.id).from(idHolderCTE)
+                    .fetch();
+
+            assertNotNull(fetch);
+        });
+    }
+
+    @Test
+    @Category({ NoDatanucleus.class, NoEclipselink.class, NoOpenJPA.class, NoMySQLOld.class, NoDB2.class })
+    public void testRecursiveCTEUnion() {
+        Assume.assumeTrue(dbmsDialect.supportsWithClause());
+
+        doInJPA(entityManager -> {
+            List<Long> fetch = new BlazeJPAQuery<Document>(entityManager, cbf)
+                    .withRecursive(idHolderCTE, idHolderCTE.id).as(unionAll(select(document.id).from(document).where(document.id.eq(1L)), select(document.id).from(document)
+                            .join(idHolderCTE).on(idHolderCTE.id.add(1L).eq(document.id))))
+                    .select(idHolderCTE.id).from(idHolderCTE)
+                    .fetch();
+
+            assertNotNull(fetch);
+        });
+    }
+
+    @Test
+    @Category({ NoDatanucleus.class, NoEclipselink.class, NoOpenJPA.class, NoMySQLOld.class })
```

Comment:
Add a comment like e.g. `// NOTE: No advanced sql support for datanucleus, eclipselink and openjpa yet` and also `// NOTE: Recursive CTE support was only added in MySQL8`.